no viable overloaded '=' in weights.hpp:127:14

Hi folks,

I'm working on a port of Gecode to Wasm using Mscripten. Mscripten generates a static library.
While building the library, I get the following error  : 
`set/int/weights.hpp:127:14: error: no viable overloaded '='

    elements = elements0; weights = weights0;

kernel/data/shared-array.hpp:53:9: note: candidate function (the implicit copy assignment operator) not viable: 'this' argument has type 'const SharedArray<int>', but method is not marked const
`
directly followed by :
`set/int/weights.hpp:127:35: error: no viable overloaded '='

    elements = elements0; weights = weights0;

kernel/data/shared-array.hpp:53:9: note: candidate function (the implicit copy assignment operator) not viable: 'this' argument has type 'const SharedArray<int>', but method is not marked const
`
And thus the build stops

**To Reproduce**

I followed this How to for Emscripen (built from the github repo) : 
https://willowtreeapps.com/ideas/how-to-use-native-libraries-on-node-js-with-emscripten

**Gecode and Platform Configuration**
Gecode 6.2.0
Linux Mint 19.2
Kernel : 4.15.0-96-generic
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 1.39.13
clang version 11.0.0 (/b/s/w/ir/cache/git/chromium.googlesource.com-external-github.com-llvm-llvm--project 9a39d5a2ecf7c9290d334f2697c3a4e533aa4bea)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/moi/ST/emsdk/upstream/bin
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/8
Selected multilib: .;@m64

** Solution (Good or not) I have found **
The problem is located in the init method that tries to override a const SharedArray. Removing the const in the definition of both elements and weights allows the build to go on. I'm not good enough in C++ to know if this solution is good or not.

--- a/gecode/set/int/weights.hpp
+++ b/gecode/set/int/weights.hpp
@@ -49,9 +49,9 @@ namespace Gecode { namespace Set { namespace Int {
     /// The value iterator
     I iter;
     /// A superset of the elements found in the iterator
-    const SharedArray<int> elements;
+    SharedArray<int> elements;
     /// Weights for all the possible elements
-    const SharedArray<int> weights;
+    SharedArray<int> weights;
     /// The current index into the elements and weights
     int index;
     /// Move to the next element

